File: data-structures/trees/binary-trees/avl-tree/cpp/avl-tree-main.cpp
#include <bits/stdc++.h>
#include "avl-tree.h"
using namespace std;

int main() {
  int option = -1;
  int val;
  AVLTree tree;
  

  while(option != 0) {
    cout << "Which operation you wanna do?" << endl;
    cout << "1. Insert node" << endl;
    cout << "2. Search node" << endl;
    cout << "3. Delete node" << endl;
    cout << "4. Print values in order" << endl;
    cout << "5. Print values pre order" << endl;
    cout << "6. Print values post order" << endl;
    cout << "7. Print tree rotated 90º counterclockwise" << endl;
    cout << "0. Exit" << endl;
    cout << "Select Option number: " << endl;
    cin >> option;

    TreeNode *node = new TreeNode();
    switch (option) {
    case 0:
      cout << "\nExiting...\n" << endl;
      break;
    case 1:
      cout << "Pass an integer value to insert:" << endl;
      cin >> val;
      cout << "\nInserting...\n" << endl;
      node->value = val;
      tree.root = tree.insertRecursive(tree.root, node);
      // code
      break;
    case 2:
      cout << "Pass a node to search:" << endl;
      cin >> val;
      cout << "\nSearching..." << endl;
      if(tree.contains(val))
        cout << "Value found!\n" << endl;
        
      else
        cout << val << "is not present in the tree.\n" << endl;
      // code
      break;
    case 3:
      cout << "Pass a node to delete:" << endl;
      cin >> val;
      if(tree.contains(val)) {
        cout << "\nDeleting...\n" << endl;
        tree.erase(tree.root, val);
      }
      else {
        cout << "\nValue not found\n" << endl;
      }
      break;
    case 4:
      cout << "\n";
      tree.printInOrder();
      cout << "\n";
      break;
    case 5:
      cout << "\n";
      tree.printPreOrder();
      cout << "\n";
      break;
    case 6:
      cout << "\n";
      tree.printPostOrder();
      cout << "\n";
      break;
    case 7:
      cout << "\n";
      tree.printCounterClockwise(tree.root, 5);
      cout << "\n";
      break;
    default:
      cout << "\nEnter proper option number\n" << endl;
      break;
    }
   
  }
  
  return 0;   
}
  

File: data-structures/trees/binary-trees/avl-tree/py/avl_tree_main.py
import avl_tree

def main():
  option = -1
  tree = AVLTree()

  while option != 0:
    print("Which operation you wanna perform?")
    print("1. Insert node")
    print("2. Search node")
    print("3. Delete node")
    print("4. Print values in order")
    print("5. Print values pre order")
    print("6. Print values post order")
    print("7. Print tree rotated 90 degrees counterclockwise")
    print("0. Exit")
    option = int(input("Select Option number: "))

    node = TreeNode();
    if option == 0:
      print("\nExiting...\n", end='')
    elif option == 1:
      val = int(input("Pass an integer value to insert: "))
      print("\nInserting {}...".format(val))
      node.value = val
      tree.root = tree.insert(tree.root, node)
    elif option == 2:
      val = int(input("Pass an integer value to search: "))
      print("\nSearching...\n", end='')
      if tree.contains(val, tree.root):
        print("Value found!\n", end='')
      else:
        print("{} is not present in the tree.\n\n".format(val), end='')
    elif option == 3:
      val = int(input("Pass a node to delete:"))
      if tree.contains(val, tree.root):
        print("\nDeleting...\n")
        tree.root = tree.deleteNode(tree.root, val)
      else:
        print("\nValue not found\n")
      # delete node
    elif option == 4:
      print("\n", end='')
      print("In order:")
      tree.printInOrder()
      print("\n", end='')
      # print values
    elif option == 5:
      print("\n", end='')
      print("Pre order:")
      tree.printPreOrder()
      print("\n", end='')
      # print values
    elif option == 6:
      print("\n", end='')
      print("Post order:")
      tree.printPostOrder()
      print("\n", end='')
      # print values
    elif option == 7:
      print("--------------------------------------------------------")
      print("Tree rotated 90 degrees counterclockwise:\n\n", end='')
      tree.printCounterClockwise(tree.root, 5)
      print("\n\n--------------------------------------------------------\n")
      # print values
    else:
      print("\nEnter proper option number\n", end='')


if __name__ == "__main__":
  main()

